tests/multiple_hooks_modify_exception.phpt

```diff
@@ -0,0 +1,21 @@
+--TEST--
+Check if hooks receive modified exception
+--EXTENSIONS--
+otel_instrumentation
+--FILE--
+<?php
+\OpenTelemetry\Instrumentation\hook(null, 'helloWorld', post: fn(mixed $object, array $params, mixed $return, Exception $throwable) => throw new Exception(previous: $throwable));
+\OpenTelemetry\Instrumentation\hook(null, 'helloWorld', post: fn(mixed $object, array $params, mixed $return, Exception $throwable) => throw new Exception(previous: $throwable));
```

Comment:
We don't update `$throwable` (also throwing in posthook might be broken in general); 
https://github.com/open-telemetry/opentelemetry-php-instrumentation/blob/f2c1a1e87ef540f2d70e2d3bdc990a32f04307ec/otel_observer.c#L263-L266